File: src/main/java/ru/timakden/chesspuzzle/Bishop.java
package ru.timakden.chesspuzzle;

/**
 * Слон, перемещается на любое число полей по диагонали.
 */
class Bishop extends ChessPiece {
    Bishop(int row, int column) {
        super(row, column);
    }

    @Override
    public String toString() {
        return "B";
    }

    public boolean canAttackAnotherChessPiece(ChessPiece anotherChessPiece) {
        int row = getRow();
        int column = getColumn();
        int anotherRow = anotherChessPiece.getRow();
        int anotherColumn = anotherChessPiece.getColumn();

        return Math.abs(row - anotherRow) == Math.abs(column - anotherColumn);
    }
}

File: src/main/kotlin/ru/timakden/chesspuzzle/Bishop.kt
package ru.timakden.chesspuzzle

/**
 * Слон, перемещается на любое число полей по диагонали.
 */
class Bishop(row: Int, column: Int) : ChessPiece(row, column) {
    override fun toString() = "B"

    override fun canAttackAnotherPiece(anotherPiece: ChessPiece): Boolean {
        return Math.abs(row - anotherPiece.row) == Math.abs(column - anotherPiece.column)
    }
}
